src/main/java/wolox/training/models/Book.java

```diff
@@ -0,0 +1,149 @@
+package wolox.training.models;
+
+import java.util.Objects;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+@Entity
```

Comment:
You can add the @NotNull annotation (import javax.validation.constraints.NotNull)  on top of each field that has nulleable = false. This is to check if the field is null or not before persisting. The verification is done in the java layer when instantiating and setting the object.